pkg/config/model.go

```diff
@@ -328,6 +330,82 @@ func DeepCopy(s any) any {
 	return data
 }
 
+func (c *Config) Equals(other *Config) bool {
```

Comment:
Gonna have to come back to this and think really deeply about it. Not that I think it's not correct, but it's pretty criticial